mne/preprocessing/ica.py

```diff
@@ -663,26 +663,26 @@ def _compute_pre_whitener(self, data):
             for ch_type in _DATA_CH_TYPES_SPLIT + ('eog', "ref_meg"):
                 if _contains_ch_type(info, ch_type):
```

Comment:
maybe there is a simpler way. We have access to the picked channels (channels on which ICA will be applied). So here for whitening we need to get the picked indices by channel types and select the picks_ from the "picked channels". Does it make sense?

Reply:
Let's see if I got it right. The channels on which the ICA will be applied are accessible in `._compute_pre_whitener` in `info = self.info`, info instance created by selecting the channels passed in arg `picks` in the method `.fit`. So now, for each channel type contained in this info instance, we need to compute the associated`pre_whitener`; essentially replacing this long list of if/else statements with a compact cleaner implementation. If that is what you want, i.e. to simplify the code and get rid of the long list of if/else statements, then sure I can look into it. I didn't do it to minimize the amount of code changed. If that's not what you meant, then I'm clueless.